File: 백준/Bronze/10870. 피보나치 수 5/피보나치 수 5.js
const input = parseInt(require("fs").readFileSync("/dev/stdin").toString().trim());
const cache = new Map();

const fibo = (input) => {
    if (input === 0) return 0;
    if (input === 1) return 1;
    if (!cache.has(input)) cache.set(input, fibo(input - 1) + fibo(input - 2));

    return cache.get(input);
}

console.log(fibo(input));

File: 백준/Bronze/10870. 피보나치 수 5/피보나치 수 5.cc
#include <stdio.h>

int fibo(int i)
{
  if (i == 0)
    return 0;
  if (i == 1)
    return 1;

  return fibo(i - 1) + fibo(i - 2);
}

int main()
{
  int n;
  scanf("%d", &n);
  printf("%d", fibo(n));
  return 0;
}